kubewarden/audit-scanner
# Audit scanner MVP1

## Description

The program takes as input the name of the `Namespace` to be inspected, then print the policies that will be audited.

```
Usage:
  audit-scanner [flags]

Flags:
  -h, --help               help for audit-scanner
  -l, --loglevel string    level of the logs. Supported values are: [trace debug info warn error fatal] (default "info")
  -n, --namespace string   namespace to be evaluated
```

It is using [zerolog](https://github.com/rs/zerolog) for the structured logging. We can easily change it if you prefer to use another library. I also considered using [zap](https://pkg.go.dev/go.uber.org/zap), but I find zerolog easier to work with. Let me know what you think

## Test
- run `make` to create the binary. It will use the `kubeconfig` in your home dir if it is not deployed in a Kubernetes cluster.
Make sure you have the latest CRD from the [feat-audit](https://github.com/kubewarden/kubewarden-controller/tree/feat-audit) branch deployed in your cluster, otherwise `backgroundAudit` will always be **false**.
- deploy `ClusterAdmissionPolicies` and `AdmissionPolicies` in your cluster
- run `./audit-scanner -n namespace -l trace ` loglevel (-l) is optional, but it is useful to set it to trace when testing. It will show which policies were skipped from being audited, and the reason.
- unit-tests can be run with `make test`


Fix https://github.com/kubewarden/audit-scanner/issues/1
